source/common/upstream/thread_aware_lb_impl.h

```diff
@@ -26,6 +28,17 @@ class ThreadAwareLoadBalancerBase : public LoadBalancerBase, public ThreadAwareL
   public:
     virtual ~HashingLoadBalancer() = default;
     virtual HostConstSharedPtr chooseHost(uint64_t hash, uint32_t attempt) const PURE;
+    const std::string hashKey(HostConstSharedPtr host, bool use_hostname) {
+      std::string hash_key = Config::Metadata::metadataValue(
+                                 host->metadata().get(), Config::MetadataFilters::get().ENVOY_LB,
+                                 Config::MetadataEnvoyLbKeys::get().HASH_KEY)
+                                 .string_value();
```

Comment:
What does this return if the value is not a string? Empty string? Should this be configuration checked somewhere? Or at least have a TODO for that? Please have a test for a non-string key.

Reply:
id imagine the yaml parser into protobuf would leave the string value empty in that case and it will get handled by the default case (fallback to hostname/resolved address)

Reply:
I think that's OK, but do you have a test for this case?

/wait

Reply:
there are tests for when the metadata is not set and the string is empty e.g. in [maglev ](https://github.com/envoyproxy/envoy/blob/main/test/common/upstream/maglev_lb_test.cc#L101-L169).

i can add another one in HashingLoadbalancer by creating a host with an int_value in the metadata instead of [string_value](https://github.com/envoyproxy/envoy/pull/14162/files#diff-1cf1654ea1233e6793e9a924bbb6ec36beff732bb7ea2c0230d9cc74b87e55f1R128) if you like. But it seems a bit redundant.

Reply:
I'm looking for a test in which metadata is set to something that is not a string, as you point out. I don't think this is redundant and it makes sure we don't see any strange behavior with what might be a common config mistake. Thank you!

/wait